WooCommerce/Classes/ViewRelated/MainTabBarController.swift

```diff
@@ -573,12 +573,34 @@ private extension MainTabBarController {
         guard featureFlagService.isFeatureFlagEnabled(.backgroundProductImageUpload) else {
             return
         }
-        productImageUploadErrorsSubscription = productImageUploader.errors.sink { [weak self] update in
-            self?.handleBackgroundImageUploadError(update)
+        productImageUploadErrorsSubscription = productImageUploader.errors.sink { [weak self] error in
+            guard let self = self else { return }
+            switch error.error {
+            case .savingProductImages:
+                self.handleImagesSaverError(error)
+            case .actionHandler:
+                self.handleBackgroundImageUploadError(error)
+            }
         }
     }
 
-    func handleBackgroundImageUploadError(_ update: ProductImageUploadError) {
+    func handleImagesSaverError(_ error: ProductImageUploadErrorInfo) {
```

Comment:
If we end up renaming the `ProductImageUploaderError` cases we could consider renaming these helper methods to match error names.

Reply:
Updated in https://github.com/woocommerce/woocommerce-ios/pull/7154/commits/eeab67f1d097fa04e1f895c2a6d50bb09e014002